lxd/backup/backup.go

```diff
@@ -0,0 +1,255 @@
+package backup
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/lxc/lxd/lxd/db"
+	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// InstanceLoadByID returns instance config by ID.
+var InstanceLoadByID func(s *state.State, id int) (Instance, error)
+
+// Instance represents the backup relevant subset of a LXD instance.
+type Instance interface {
+	Name() string
```

Comment:
Hmm, how come we don't have `Project()` in there, surely that's needed to properly locate backups?

Or are backups completely broken with projects right now?

Reply:
Ouch, yeah, looking below, looks like we have a bunch of places where we should be using project.Prefix() in on-disk paths to avoid clashes/issues with the same container name in multiple projects.

Can you push a follow-up commit in this branch which sorts that out and also adds a test for this?

Reply:
Done